### PR TITLE
fix: adding new property to tagged elements that are overlapped by others [TOL-2408]

### PR DESCRIPTION
**Pull request by aodhagan-cf**
Checking intersections of overlapping tagged elements and adding an attribute which is sent to the editor to stop rendering tagged elements which should be hidden.

- For each 4 corners of the tagged elements calculate what the topmost element is at that point
- If the topmost element is not the tagged element add property to hide tagged element

### CODE REVIEW

File: packages/live-preview-sdk/src/inspectorMode/index.ts [hunk 1]
```diff
@@ -239,10 +235,11 @@ export class InspectorMode {
       {
         elements: this.taggedElements.map((taggedElement) => ({
           // Important: do not add `element` as it can't be cloned by sendMessage
-          coordinates: taggedElement.coordinates,
-          isVisible: taggedElement.isVisible,
+          coordinates: taggedElement.coordinates!,
+          isVisible: !!taggedElement.isVisible,
```
Comment (chrishelgert) [inline packages/live-preview-sdk/src/inspectorMode/index.ts:239]: Could we fix the typings when we create the taggedElements?

**Review by chrishelgert — APPROVED** [review]

File: packages/live-preview-sdk/src/inspectorMode/index.ts (hunk 1 above)
Comment (aodhagan-cf) [reply packages/live-preview-sdk/src/inspectorMode/index.ts:239]: I've added a new type now

PrecaulculatedTaggedElement 

This just has the stuff that you can get directly from the tagged element, so element + attributes

The the actual TaggedElement has the calculated state without any conditional properties

File: packages/live-preview-sdk/src/inspectorMode/utils.ts [hunk 2]
```diff
@@ -271,3 +281,53 @@ export function getAllTaggedEntries({
...
+  }));
+
+/**
+ * Applies the attributes that we cannot simply get from the tagged elements itself
+ * but need to calculate based on the current state of the document
+ */
+export const addCalculatedAttributesToTaggedElements = (
+  taggedElements: PrecaulculatedTaggedElement[],
+  root = window.document,
+): TaggedElement[] => {
```
Comment (chrishelgert) [inline packages/live-preview-sdk/src/inspectorMode/utils.ts:330]:
```suggestion
): PrecaulculatedTaggedElement[] => {
```

Comment (chrishelgert) [reply packages/live-preview-sdk/src/inspectorMode/utils.ts:330]: same then in 332

Comment (aodhagan-cf) [reply packages/live-preview-sdk/src/inspectorMode/utils.ts:330]: No this should use the regular one as it now has added the calculations

Comment (aodhagan-cf) [reply packages/live-preview-sdk/src/inspectorMode/utils.ts:330]: I've merged for now, let me know I can follow up later on this if you disagree

**Pull request merged**